### ISSUE TITLE
RequestStream does not close stream when buffering to memory

### ISSUE THREAD

**Issue by tekhedd** [NancyFx/Nancy#1986]
MoveToWritableStream() does not close the old stream after copying it to memory. This leaves HttpListenerRequest's stream unclosed when self-hosting.

**Comment by phillip-haydon** [comment]
Is this for self hosting without OWIN? 

Could we just drop support for non-owin based self hosting?

**Comment by tekhedd** [comment]
This is for self-hosting. Are you suggesting that the best fix to the bug I reported that I found while implementing non-OWIN based SelfHosting in my app is to drop support? Awesome. :}

**Comment by phillip-haydon** [comment]
@tekhedd no. I'm suggesting dropping support for self hosting in Nancy in favour of Owin self hosting which doesn't suffer from this problem. I personally don't know anyone not using non-owin self hosting now-a-days. The old self hosting has other issues besides the one you have raised. 

Is there any reason why you can't use the owin host?

**Comment by tekhedd** [comment]
1) I'm running cross platform here. A stable, widely used OWIN implementation that runs on Mono... would be...which one again? 

2) I personally don't know anyone who watches soccer. Why do they keep broadcasting it on the TV?

**Comment by khellang** [comment]
Katana is both widely used and stable. What's the problem on Mono?
